### ISSUE TITLE
Idea: Assign Scale and Expo shortcuts to mouse buttons

### ISSUE THREAD

**Issue by wilecoyote2015** [linuxmint/Cinnamon#861]
Cinnamon does not support mouse buttons as shortcuts yet. 
It would enhance productivity if you could assign the Scale and Expo effects to the mouse buttons, too because that would provide more direct and faster access on multi button mouses.

**Comment by haight6716** [comment]
+1 switching back to unity because of this..

**Comment by simonwiles** [comment]
@cast-away -- *can* this actually be achieved with `xbindkeys`?  on a cursory attempt I couldn't get it to recognize mouse buttons...

**Comment by dzarafata** [comment]
+1, that would be very handy

**Comment by jiminikiz** [comment]
+1

**Comment by cast-away** [comment]
1. Install required packages.
`sudo apt-get install xev xbindkeys`

2. Find your mouse button number.
run xev in terminal, move the cursor over the square in the new window and press the desired button, the output will be something like this:
```
ButtonPress event, serial 38, synthetic NO, window 0x3c00001,
    root 0x25b, subw 0x3c00002, time 9084960, (45,47), root:(970,538),
    state 0x10, button 9, same_screen YES
```
3. Link mouse button (9 as in previous output) to keyboard shortcut (Crl+Alt+Down) with xbindkeys.
`gedit ~.xbindkeysrc`
and add the following:
```
"xte 'keydown Control_L' 'keydown Alt_L' 'key Down' 'keyup Alt_L' 'keyup Control_L'"
b:13 + Release
```
4. Make sure the entry `xbindkeys_autostart` is present in your startup applications.

5. Log in and log out for the changes to take effect.

**Comment by KaloyanKaludov** [comment]
+1

**Comment by abdupattoh** [comment]
++

**Comment by tobia** [comment]
This would indeed be useful.

I solved my use case with a slightly different xbindkeys config, that calls dbus-send:

    "dbus-send --session --type=method_call --dest=org.Cinnamon /org/Cinnamon org.Cinnamon.ShowExpo"
    b:8 + Release

The `+ Release` part was the key to make it work, for some reason. I found the name of the DBus method by using d-feet.

Finally, I'm not yet sure where the startup applications are stored, I just used the Startup Applications menu entry to check that xbindkeys had registered itself.
